source/transactions/tests/unified/pin-mongos.yml

```diff
@@ -0,0 +1,646 @@
+# Test that all the operations go to the same mongos.
+#
+# In tests that don't include command-started events the assertion is implicit:
+# that all the read operations succeed. If the driver does not properly pin to
+# a single mongos then one of the operations in a transaction will eventually
+# be sent to a different mongos, which is unaware of the transaction, and the
+# mongos will return a command error. An example of such an error is:
+# {
+#   'ok': 0.0,
+#   'errmsg': 'cannot continue txnId -1 for session 28938f50-9d29-4ca5-8de5-ddaf261267c4 - 47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU= with txnId 1',
+#   'code': 251,
+#   'codeName': 'NoSuchTransaction',
+#   'errorLabels': ['TransientTransactionError']
+# }
+description: pin-mongos
+
+schemaVersion: '1.9'
+
+runOnRequirements:
+  -
+    minServerVersion: 4.1.8
+    # serverless proxy doesn't append error labels to errors in transactions
+    # caused by failpoints (CLOUDP-88216)
+    serverless: forbid
+    topologies:
+      - sharded
+      - load-balanced
```

Comment:
These tests don't actually pass against a load balancer since you can't target a fail point through one.  This slipped through in the legacy tests because the Java test runner, which we used for verification, skipped the tests and I didn't notice.  

I suggest we revert the addition of load-balanced to at least some of the legacy tests that were added in scope of DRIVERS-2713.  And then change this PR to do the same in the unified tests.

Not sure if they all need to be reverted, but at least [the ones that were skipped](https://spruce.mongodb.com/task/mongo_java_driver_tests_load_balancer_secure__version~7.0_os~ubuntu_topology~sharded_cluster_auth~auth_ssl~ssl_jdk~jdk17_load_balancer_test_patch_cb6692fdea4e231b51fd9ec3c7c9380f9eb84f45_65c533cb56234390ce6e04da_24_02_08_20_04_58/tests?execution=0&page=0&sortBy=STATUS&sortDir=ASC&statuses=skip&testname=com.mongodb.client.MainTransactionsTest) in the original patch build for the Java driver.

Reply:
> I suggest we revert the addition of load-balanced to at least some of the legacy tests that were added in scope of [DRIVERS-2713](https://jira.mongodb.org/browse/DRIVERS-2713).

I don't see the point in reverting any changes to legacy tests modified by DRIVERS-2713, since this PR deletes all of those tests.

> And then change this PR to do the same in the unified tests. Not sure if they all need to be reverted, but at least [the ones that were skipped](https://spruce.mongodb.com/task/mongo_java_driver_tests_load_balancer_secure__version~7.0_os~ubuntu_topology~sharded_cluster_auth~auth_ssl~ssl_jdk~jdk17_load_balancer_test_patch_cb6692fdea4e231b51fd9ec3c7c9380f9eb84f45_65c533cb56234390ce6e04da_24_02_08_20_04_58/tests?execution=0&page=0&sortBy=STATUS&sortDir=ASC&statuses=skip&testname=com.mongodb.client.MainTransactionsTest) in the original patch build for the Java driver.

I opened [DRIVERS-2834](https://jira.mongodb.org/browse/DRIVERS-2834) to track this, and also suggested evaluating the `runOnRequirements` for `withTransaction` tests in that issue.

Reply:
Done.

Reply:
@jyemin: I added a commit that excludes `load-balanced` for any tests using `targetedFailPoint`. Please let me know if that allows you to remove all of the manual skips in Java. I'll test this with PHPLIB in the meantime, but I don't think any of these tests were running on LB to begin with.